python/cucim/src/cucim/skimage/morphology/convex_hull.py

```diff
@@ -0,0 +1,315 @@
+        for (int i_eq = 0; i_eq < n_hull_equations; i_eq++) {"""
+    source += f"""
+          {float_t} v = 0.0;"""
+    for d in range(ndim):
+        source += f"""
+          v += hull_equations[i_eq * {ndim + 1} + {d}] * in_coord[{d}];"""
+    source += f"""
+          v += hull_equations[i_eq * {ndim + 1} + {ndim}];"""
+    source += """
+          if (v > tol) {
+            in_hull = false;
+            break;
+          }
+        }
+        convex_image = in_hull;
+      }\n"""
+    inputs = (
+        f"raw bool image, raw {float_dtype.name} hull_equations, float64 tol"
+    )
+    outputs = "bool convex_image"
+    name = f"cucim_convex_hull_{ndim}d_{coord_dtype.char}_{float_dtype.char}"
+    return cp.ElementwiseKernel(inputs, outputs, source, name=name)
+
+
+def convex_hull_image(
+    image,
+    offset_coordinates=True,
+    tolerance=1e-10,
+    include_borders=True,
+    *,
+    omit_empty_coords_check=False,
+    float64_computation=True,
+    cpu_fallback_threshold=None,
+):
+    """Compute the convex hull image of a binary image.
+
+    The convex hull is the set of pixels included in the smallest convex
+    polygon that surround all white pixels in the input image.
+
+    Parameters
+    ----------
+    image : array
+        Binary input image. This array is cast to bool before processing.
+    offset_coordinates : bool, optional
+        If ``True``, a pixel at coordinate, e.g., (4, 7) will be represented
+        by coordinates (3.5, 7), (4.5, 7), (4, 6.5), and (4, 7.5). This adds
+        some "extent" to a pixel when computing the hull.
+    tolerance : float, optional
+        Tolerance when determining whether a point is inside the hull. Due
+        to numerical floating point errors, a tolerance of 0 can result in
+        some points erroneously being classified as being outside the hull.
+    include_borders: bool, optional
+        If ``False``, vertices/edges are excluded from the final hull mask.
+
+
+    Extra Parameters
+    ----------------
+    omit_empty_coords_check : bool, optional
+        If ``True``, skip check that there are not any True values in `image`.
+    float64_computation : bool, optional
+        If False, allow use of 32-bit float during the postprocessing stage
+        that determines whether each pixel falls within the convex hull.
+    cpu_fallback_threshold : non-negative int or None
+        Number of pixels in an image before convex_hull_image will fallback
+        to pure CPU implementation.
+
+    Returns
+    -------
+    hull : (M, N) array of bool
+        Binary image with pixels in convex hull set to True.
+
+    Notes
+    -----
+    The parameters listed under "Extra Parameters" above are present only
+    in cuCIM and not in scikit-image.
+
+    References
+    ----------
+    .. [1] https://blogs.mathworks.com/steve/2011/10/04/binary-image-convex-hull-algorithm-notes/
+
+    """
+    if cpu_fallback_threshold is None:
```

Comment:
```suggestion
    if tolerance <= 0:
        raise ValueError("tolerance must be positive")   
    if cpu_fallback_threshold is None:
```

Reply:
Thanks, I think this tolerance check is not present on the scikit-image side, but we can still add it